openfl/transport/grpc/director_server.py

```diff
@@ -224,20 +227,37 @@ async def RemoveExperimentData(self, request, context):  # NOQA:N802
         response.acknowledgement = True
         return response
 
-    async def CollaboratorHealthCheck(self, request, context):  # NOQA:N802
+    async def EnvoyHealthCheck(self, request, context):  # NOQA:N802
         """Accept health check from envoy."""
-        logger.debug(f'Request CollaboratorHealthCheck has got: {request}')
-        health_check_period = self.director.collaborator_health_check(
-            collaborator_name=request.name,
+        logger.debug(f'Request EnvoyHealthCheck has got: {request}')
+        cuda_devices_info = [
+            MessageToDict(message, preserving_proto_field_name=True)
+            for message in request.cuda_devices
+        ]
+        health_check_period = self.director.envoy_health_check(
+            envoy_name=request.name,
             is_experiment_running=request.is_experiment_running,
+            cuda_devices_status=cuda_devices_info
         )
-        resp = director_pb2.CollaboratorHealthCheckResponse()
+        resp = director_pb2.EnvoyHealthCheckResponse()
         resp.health_check_period.seconds = health_check_period
 
         return resp
 
     async def GetEnvoys(self, request, context):  # NOQA:N802
         """Get a status information about envoys."""
         envoy_infos = self.director.get_envoys()
-
-        return director_pb2.GetEnvoysResponse(envoy_infos=envoy_infos)
+        response = []
```

Comment:
It's not a response. It's envoy_infos or envoy_info_messages. Response is director_pb2.GetEnvoysResponse